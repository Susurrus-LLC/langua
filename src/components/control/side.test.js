import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import ControlSide from './Side'

describe('ControlSide', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ControlSide />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately with and without props and children', () => {
    let tree = renderer.create(<ControlSide />).toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderer.create(<ControlSide addedClasses='test' />).toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderer.create(<ControlSide>Test</ControlSide>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
