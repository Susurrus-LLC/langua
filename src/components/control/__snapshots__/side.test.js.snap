// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlSide renders accurately with and without props and children 1`] = `
<div
  className="controlSide controlRight"
/>
`;

exports[`ControlSide renders accurately with and without props and children 2`] = `
<div
  className="controlSide controlRight test"
/>
`;

exports[`ControlSide renders accurately with and without props and children 3`] = `
<div
  className="controlSide controlRight"
>
  Test
</div>
`;
